EightBall.java

```diff
@@ -0,0 +1,17 @@
+package com.fortify.sca.plugins.maven.samples;
+
+import java.io.FileReader;
+
+public class EightBall {
+	public static void main(String args[]) throws Exception {
+		char[] buffer = new char[1024];
+		String filename = args[0];
+		try {
+			filename = "" + (Integer.parseInt(filename) % 3);
+		} catch (Exception e) {
+			System.out.println("Invalid input.");
```

Comment:
<p><b>Security Scanning</b> / Fortify SAST</p>
<h3>High - Unreleased Resource: Streams </h3>
<p>The function main() in EightBall.java sometimes fails to release a system resource allocated by FileReader() on line 12.


</p>
<br>
<p><a href=http://localhost:2320/ssc/html/ssc/version/46/fix/7249/?engineType=SCA&issue=E4B3ACF92911ED6D98AAC15876739EC7&filterSet=f2f88863-0d8f-4d22-b001-3e1cd3ec0f45 target="_blank" rel="noopener noreferrer">More detailed information</a></p>